### ISSUE TITLE
Annotation cleaner runs an expensive cleanup query when configured to clean nothing

### ISSUE THREAD

**Issue by afayngelerindbx** [grafana/grafana#33948]
**What happened**:
We are running Grafana configured with default annotation cleanup options(`max_age` and `max_annotation_age` is not configured in any of the appropriate sections). However, periodically, we encounter availability hits due to e.g. annotations API  endpoints timing out, even saturating serving processes causing failed health checks.

These availability hits were traced to the following query archetype holding locks on `annotation_tags`: 
```
DELETE FROM annotation_tag WHERE id IN ( SELECT id FROM (SELECT id FROM annotation_tag WHERE NOT EXISTS (SELECT 1 FROM annotation a WHERE annotation_id = a.id)  LIMIT 100)
```

**What you expected to happen**:
No annotation cleanup queries. No availability hits :)

Looking at the code, there is no way to configure this query to not be run. I don't understand Grafana well enough to stipulate all scenarios where a row in `annotation_tag` becomes orphaned. However, if the only scenario that this query accounts for is the annotations cleanup, there's a trivial fix: don't run the `annotation_tag` [cleanup](https://github.com/grafana/grafana/blob/main/pkg/services/sqlstore/annotation_cleanup.go#L51)  when `totalCleanedAnnotations == 0`

If that is all that's needed to fix this, I'd be happy to submit a patch.

**How to reproduce it (as minimally and precisely as possible)**:
To repro you'd have to generate a fairly large `annotation_tags` table and a moderate amount of annotations API traffic.

**Anything else we need to know?**:

**Environment**:
- Grafana version: 7.5-beta3
- Data source type & version: custom
- OS Grafana is installed on: Ubuntu 20
- User OS & Browser: All
- Grafana plugins: lots
- Others: mySQL

**Comment by marefr** [comment]
Thanks for reporting. This seems wrong. Seems to have been introduced by #29534 

> don't run the annotation_tag cleanup when totalCleanedAnnotations == 0

I think that fix makes sense. So if you want, please let us know if you're planning to submit a patch. Thanks

**Comment by afayngelerindbx** [comment]
I'll get a patch up today.

**Comment by afayngelerindbx** [comment]
What's the process for requesting a back port of this fix onto a 7.5 point release?
